### PR TITLE
Log locations received from Mapbox and how they change over time

### PR DESCRIPTION
**Pull request by lawrence-forooghian**
I wrote this to help in the investigation of #377, but I think it'll come in useful in the future too so let’s keep it around. See commit messages for more details.

### CODE REVIEW

**Review by KacperKluka — COMMENTED** [review]
Overall it looks good but I'm wondering whether this might impact the battery and introduce some delays in location processing :thinking: Maybe it would be better to add some kind of a debug flag to the SDK and only do those calculations and logging when the flag is enabled? WDYT?

File: Sources/AblyAssetTrackingPublisher/Services/DefaultLocationService.swift [hunk 1]
```diff
@@ -53,6 +55,17 @@ extension DefaultLocationService: PassiveLocationManagerDelegate {
     }
     
     func passiveLocationManager(_ manager: PassiveLocationManager, didUpdateLocation location: CLLocation, rawLocation: CLLocation) {
+        logHandler?.verbose(message: "\(String(describing: Self.self)), passiveLocationManager(\(manager), didUpdateLocation: \(String(reflecting: location)), rawLocation: \(String(reflecting: rawLocation)))", error: nil)
+
+        if let lastReceivedLocations = lastReceivedLocations {
+            let locationDifferences = LocationDifferenceCalculator.calculateDifferences(first: location, second: lastReceivedLocations.location)
+            logHandler?.verbose(message: "\(String(describing: Self.self)), Differences between location and last received location: \(String(reflecting: locationDifferences))", error: nil)
+            let rawLocationDifferences = LocationDifferenceCalculator.calculateDifferences(first: rawLocation, second: lastReceivedLocations.rawLocation)
+            logHandler?.verbose(message: "\(String(describing: Self.self)), Differences between rawLocation and last received rawLocation: \(String(reflecting: rawLocationDifferences))", error: nil)
```
Comment (ikbalkaya) [inline Sources/AblyAssetTrackingPublisher/Services/DefaultLocationService.swift:64]: Shouldn't this and above be `debug` rather than `verbose`?

**Review by ikbalkaya — COMMENTED** [review]
I think it's a good idea to have the `LocationDifferentCalculator` for debugging, but as far as I can see it's being used regardless of whether the app is in debug mode. 
I also think that we are debugging the behaviour of Mapbox SDK. IMO, it would be cleaner if we get a clean answer from them about the sending of exact locations in different timestamps (I will ask them) and then we can still use this if we really need a utility that computes differences between locations

**Comment by lawrence-forooghian** [comment]
Putting this into draft until we hear back on https://github.com/mapbox-collab/Ably-DH-Collab/issues/16.

**Comment by lawrence-forooghian** [comment]
We don't need this at the moment. Closing.

**Pull request closed without merging**